File: clean_it/src/main/java/com/example/clean_it/model/Order.java
package com.example.clean_it.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "orders")
public class Order {

    @Id
    private String id;  // MongoDB document ID

    private String customerId;  // Reference to the Customer
    private List<TextileItem> items;  // List of clothes submitted for cleaning
    private String status;  // Could be "submitted", "cleaned", "returned"

}

File: clean_it_frontend/clean_it_front/src/app/models/order.ts
export class TextileItem {
  type: string = '';  // Provide default value
  status: string = '';  // Provide default value

  constructor(type: string = '', status: string = '') {
    this.type = type;
    this.status = status;
  }
}

export class Order {
  id?: string;  // Optional for when creating a new order
  customerId: string | undefined = '';  // Provide default value
  items: TextileItem[] = [new TextileItem()];  // Default empty TextileItem
  status: string = 'submitted';  // Provide default value for status

  constructor(customerId: string = '', items: TextileItem[] = [new TextileItem()], status: string = 'submitted') {
    this.customerId = customerId;
    this.items = items;
    this.status = status;
  }
}

